examples/core-juttle/npm_download_counts.juttle

```diff
@@ -1,6 +1,55 @@
-input npm_package: text -default "juttle";
+input npm_package: text -default "juttle" -label "NPM Package";
+input start: date -default :today: - :2 years: -label "Start Date";
+input end:   date -default :today: -label "End Date";
 
-read http -url 'https://api.npmjs.org/downloads/range/last-month/${npm_package}' -timeField "day" -rootPath "downloads"
-    | put package=npm_package
-    | view timechart -keyField "package" -series [{name: npm_package, geom: 'bars'}] -yScales.primary.label "Number of NPM Downloads"
+function dayof(date) {
+    return Date.format(date, "YYYY-MM-DD");
+}
 
+// fetch the data
+sub fetch() {
+    read http -url 'https://api.npmjs.org/downloads/range/${dayof(start)}:${dayof(end)}/${npm_package}' -timeField "day" -rootPath "downloads"
+}
+
+// add a simple title using markdown
+sub title() {
+    emit -points [{title: '## Download Counts for ${npm_package}'}]
+    | view markdown
+}
+
+// Sum up the data by the given interval
+sub rollup(interval) {
+    reduce -every interval value=sum(downloads)
+    | put name='downloads'
+}
+
+// Add a rolling average series
+sub trend(window, label) {
+    (pass; reduce -every :1d: -over window value=Math.floor(avg(value)), name=label)
+    | (pass; view text)
+}
+
+sub chart(interval, label, row, col, color) {
+    view timechart
+        -keyField 'name'
+        -series [{name: 'downloads', color: color, geom: 'bars'},
+                 {name: 'trend', color: '#209542'} ]
+        -title 'NPM Downloads per ${label} for ${npm_package}'
+        -yScales.primary.label "Download Count"
+        -downsample false
+        -interval interval
+        -row row -col col
+}
+
+title;
+fetch
+| (
+    rollup -interval :1d: //| trend -window :28d: -label '28 day trend'
```

Comment:
Sorry I had a stale version pushed. Just updated with the correct version.


On Thu, Mar 10, 2016 at 5:14 PM, Daria Mehra <notifications@github.com>
wrote:

> In examples/core-juttle/npm_download_counts.juttle
> <https://github.com/juttle/juttle-engine/pull/84#discussion_r55777145>:
>
> > +sub chart(interval, label, row, col, color) {
> > +    view timechart
> > +        -keyField 'name'
> > +        -series [{name: 'downloads', color: color, geom: 'bars'},
> > +                 {name: 'trend', color: '#209542'} ]
> > +        -title 'NPM Downloads per ${label} for ${npm_package}'
> > +        -yScales.primary.label "Download Count"
> > +        -downsample false
> > +        -interval interval
> > +        -row row -col col
> > +}
> > +
> > +title;
> > +fetch
> > +| (
> > +    rollup -interval :1d: //| trend -window :28d: -label '28 day trend'
>
> the trends are commented out, didn't you want them in?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/juttle/juttle-engine/pull/84/files#r55777145>.
>



-- 
Michael Demmer
Jut - CTO / VP Engineering
demmer@jut.io